packages/docusaurus-theme-live-codeblock/src/theme/CodeBlock/index.tsx

```diff
@@ -6,17 +6,30 @@
  */
 
 import React from 'react';
+import {parseCodeBlockMetaOptions} from '@docusaurus/theme-common/internal';
 import Playground from '@theme/Playground';
 import ReactLiveScope from '@theme/ReactLiveScope';
 import CodeBlock, {type Props} from '@theme-init/CodeBlock';
 
 const withLiveEditor = (Component: typeof CodeBlock) => {
   function WrappedComponent(props: Props) {
-    if (props.live) {
-      return <Playground scope={ReactLiveScope} {...props} />;
+    const metaOptions = parseCodeBlockMetaOptions(
```

Comment:
Isn't this going to lead to parsing 3 times the metastring?

We don't want to parse it for:
- the live code block
- the code block
- the playground

We only want to parse options once

Reply:
No, I tried to ensure we parse it only once. Its a bit explained in the PR motivation: 

> Due to the way the live code editor works the metaOptions is introduced as component prop. This way we can parse it once and pass it through. The possible overlap between prop<->metaOptions is the same like prop<->metastring before.

I added a new `metaOptions` prop for this purpose to pass it from parent to child when possible. The forwarding is happening here in line 26 and 32. 

[`parseCodeBlockMetaOptions`](https://github.com/facebook/docusaurus/pull/11019/files#diff-8e027af9ea4010c1b9d4a7101a7f292283360b541235899e6799270b4030b3dfR344-R347) prioritizes the bag from the props if available.